packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -158,33 +152,37 @@ export async function associateAccountToChainArgs(
     ).sub as TypeDef[]
   )[0].type // get the type of the first key, which is the DidAddress
 
-  const encoded = api
-    .createType(`(${DidAddress}, ${BlockNumber})`, [toChain(did), validTill])
+  return api
+    .createType(`(${DidAddress}, ${BlockNumber})`, [toChain(did), validUntil])
     .toU8a()
+}
 
-  const isAccountId32 = decodeAddress(accountAddress).length > 20
-  const length = stringToU8a(String(encoded.length))
-  const paddedDetails = u8aToHex(
-    isAccountId32
-      ? u8aWrapBytes(encoded)
-      : u8aConcatStrict([U8A_WRAP_ETHEREUM, length, encoded])
-  )
-
-  const { signature, type } = getUnprefixedSignature(
-    paddedDetails,
-    await sign(paddedDetails),
-    accountAddress
-  )
+/**
+ * Generates the arguments for the extrinsic that links an account to a DID.
+ *
+ * @param accountAddress Address of the account to be linked.
+ * @param validUntil Last blocknumber that this challenge is valid for.
+ * @param signature The signature for the linking challenge.
+ * @param type The key type used to sign the challenge.
+ * @returns The arguments for the call that links account and DID.
+ */
+export async function getLinkingArguments(
+  accountAddress: Address,
+  validUntil: BN,
+  signature: Uint8Array,
+  type: KeypairType
```

Comment:
Same here, since the `type` parameter was introduced in this PR, as far as I can see, and is not already part of some user-exposed API.

Reply:
[065949a](https://github.com/KILTprotocol/sdk-js/pull/730/commits/065949a9cdb11b8825a6467e57fd31d4f3367a6c)

Reply:
Here we actually need to use something that can construct a `EncodedSignature`. 

```
  const proof = { [type]: signature } as EncodedSignature
```

the type here needs to be the signature type. So `KeypairType` is quite accurate.